### ISSUE TITLE
Display corruption and system crashes afer update to 4.1 RC4

### ISSUE THREAD

**Issue by petaramesh** [QubesOS/qubes-issues#7227]
### Qubes OS release

Qubes 4.1 RC4

### Brief summary

After a successful upgrade from a (very stable and daily used) Qubes 4.0 to 4.1 RC4, the system shows heavy graphics corruption issues and eventually crashes and reboots.

The system is a Lenovo Thinkpad P51 with hybrid Intel + NVidia graphics

### Detailed description

(I also posted this to the forum)

I had absolutely no issues in 4.0 - except that I couldn’t mount anymore EXFAT filesystems after I upgraded my Fedora VMs to Fedora 34, as it seems that Qubes 4.0 was not able to provide the correct recent kernel with EXFAT support.

So I upgraded to 4.1 RC4 - The upgrade went well.

After upgrading the good news is that I get the EXFAT support in my VMs.

Bad news is that I get a whole lot of graphics corruption and instability. (using Fedora VMs)

First I notice small visual corruption in the Plymouth splash screen during boot.

Symptoms are that with some applications, resizing windows will completely fail to resize the windows contents, making such applications unusable.

Trying to process the same contents with different apps works perfectly for reasons I don’t understand.

i.e. : Opening documents with LibreOffice shows graphics issues, opening them with OnlyOffice Desktop Editors is OK.

Opening password files with keepass shows graphics issues, opening them with keepassxc is OK.

Opening RDP sessions with Remmina shows graphics issues, opening them with xfreerdp is OK.

If trying to play several times with a window that is failing, this usually ends in the applications crashing (window disappears) but I’ve seen 4 times THE WHOLE SYSTEM SUDDENLY REBOOT.

(The hardware had been perfectly stable with Qubes 4.0).

I tried using only the NVidia graphics card by disabling the “hybrid“ feature in BIOS, and I noticed the graphics were more reactive, and no corruption in Plymouth during boot.

But with NVidia only, the problematic app stays problematic and behave differently :

When enlarging windows, the windows will redraw properly up to a given size, if enlarged further they will stop redrawing, and it continuing to fiddle with them the whole system will eventually crash.

Any clue to solve this would be greatly appreciated as having this system usable is paramount for me ; if I can’t find a solution very quickly, I will have to downgrade my system to the previous, stable, Qubes 4.0.

Thanks for having read.

**Comment by marmarek** [comment]
At least some of the issues sounds like https://github.com/QubesOS/qubes-issues/issues/7179, where the problem is identified as missing `gnttab_max_frames=2048` on the Xen cmdline.

**Comment by petaramesh** [comment]
Hello,
You're right, this really smells the same.
Unfortunately I need this machine so badly on a daily basis that I had no chance to investigate this and had to revert it back to 4.0 to have it usable again.
I may have a chance to test this in the coming couple of months, but alas definitely not in the coming days.

I also notice (now that I check) that 4.0 uses the kernel KMS driver with the Intel i915 module, whereas AFAIR in 4.1 Xorg.0.log showed no trace of the i915 module (but rather a generic KMS ?).
Still, for said reason, I cannot check this for now.

**Comment by dmoerner** [comment]
I never experienced a system crash, but if you want to revert the switch to modesetting, you might want to check out this existing bug report: https://github.com/QubesOS/qubes-issues/issues/4782#issuecomment-975920223

**Comment by petaramesh** [comment]
Hello, 
After my bad experience with 4.1 RC4 I had restored my Qubes 4.0 backup.

Today I have upgraded again to 4.1 final (using the same documented method as I previously did) and I'm happy to report that the display corruption issues are now gone.

The Plymouth boot screen still shows some glitches while typing the passphrase, and once logged I could notice that some display operation (i.e. enlarging windows) seems a bit more “flashing” than when using 4.0 where it was smooth, but the bad corruption I had in RC4 is gone.

So my system seems perfectly usable in 4.1 now, to my satisfaction.

Thanks for the good work !
